diagnostics/networking_diagnostics.bat

```diff
@@ -0,0 +1,60 @@
+:: Check for administrator privileges.
+net session >nul 2>&1 || goto :admin
+
+:: Validate that required files are here.
+if not exist wsl_networking.wprp (echo wsl_networking.wprp not found && exit /b 1)
+if not exist networking.sh (echo networking.sh not found && exit /b 1)
+
+set networking_folder=".\networking_logs"
+set neworking_logs_zip=".\WslNetworkingLogs.zip"
+
+mkdir %networking_folder%
+
+cd %networking_folder%
+
+IF "%1"=="--stop-wsl" (
+   :: The WSL HNS network is created once per boot. Resetting it to collect network creation logs.
+   echo Deleting HNS network
+   powershell.exe -NoProfile "Get-HnsNetwork | Where-Object {$_.Name -eq 'WSL'} | Remove-HnsNetwork"
+
+   :: Stop WSL.
+   net.exe stop WslService || net.exe stop LxssManager
+)
+
+wsl.exe tr -d "\r" ^| bash < ../networking.sh > wsl_network_configuration_before.log
+powershell Get-NetRoute > get_netroute.log
+
+powershell invoke-expression 'cmd /c start powershell -Command { ..\collect-wsl-logs.ps1 }'
+powershell invoke-expression 'cmd /c start powershell -Command { "wsl.exe -u root sudo tcpdump -n -i any > tcpdump.log" }'
+
+wpr -start ..\wsl_networking.wprp -filemode -instanceName wpr_networking
+pktmon start -c --flags 0x1A
+netsh wfp capture start
+
+pause
+
+:: allow some time for the user to stop logs in all the spawned shells

```

Comment:
This isn't a great UX for the user. Could we kill those processes ourselves instead ? We might need to switch to a powershell script for that, but I think it would be worth it because users will most likely get confused here

Reply:
Actually looking at this script it's probably a good idea to move it to native powershell given how much we call powershell.exe in it

Reply:
moved the script to powershell and changed it to automatically stop the log collection - now the user just needs to press a key to stop logs, similar to collect-wsl-logs.ps1